raft.go

```diff
@@ -348,12 +348,12 @@ func (rn *Node) Stop() error {
 	rn.logger.Debug("Stopping raft transporter")
 	rn.transport.Stop()
 	// TODO: Don't poll stuff here
-	for rn.running {
+	for rn.running.IsSet() {
 		time.Sleep(200 * time.Millisecond)
 	}
 	rn.logger.Info("Canoe has stopped")
-	rn.started = false
-	rn.initialized = false
+	rn.started.Unset()
+	rn.initialized.Unset()
```

Comment:
Do you rely on the ordering of these atomic bool changes? Because ordering like this on two different memory locations is not guaranteed.

Golang's memory model in the spec says:

```
That is, compilers and processors may reorder the reads and writes executed within a single goroutine only when the reordering does not change the behavior within that goroutine as defined by the language specification. Because of this reordering, the execution order observed by one goroutine may differ from the order perceived by another.
```

So you *can't* rely on the order these are set being the same between two separate goroutines running the same code potentially.

This may well not matter, but if it breaks assumptions to re-order these randomly then it's still racey.